File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",function ($http) {
    //查询所有
    this.findAll=function () {
        return $http.get("../specification/findAll.do");
    }
    //条件查询+分页

    this.search=function (searchEntity,pageNum,pageSize) {
        return $http.post("../specification/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    //根据品牌查询
    this.findOne=function (id) {
        return $http.get("../specification/findOne.do?id="+id);
    }

    //删除品牌
    this.delete=function (selectIds) {
        return $http.get("../specification/delete.do?ids="+selectIds)
    }

    //添加
    this.add=function (entity) {
        return $http.post("../specification/add.do",entity);
    }

    //修改
    this.update=function (entity) {
        return $http.post("../specification/update.do",entity);
    }

})

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.groupentity.Specification;
import com.pinyougou.pojo.TbSpecification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    //规格的分页+条件
    PageResult findPageByNameAndChar(TbSpecification specification, Integer pageNum, Integer pageSize);
    //添加规格
    void add(Specification specification);
    //根据规格id查找
    Specification findOne(Long id);

    void update(Specification specification);

    void delete(Long[] ids);
    //模板关联  查询规格表
    List<Map> selectSpecList();
}
